common/display/displayqueue.cpp

```diff
@@ -274,33 +275,26 @@ void DisplayQueue::HandleUpdateRequest(DisplayQueueItem& queue_item) {
 
   if (!display_plane_manager_->CommitFrame(current_composition_planes,
                                            pset.get(), flags)) {
-    succesful_commit = false;
-  } else {
-    display_plane_manager_->EndFrameUpdate();
-    previous_layers_.swap(queue_item.layers_);
-    previous_plane_state_.swap(current_composition_planes);
-  }
-
-  if (!succesful_commit || (flags & DRM_MODE_ATOMIC_ALLOW_MODESET))
     return;
+  }
 
+  display_plane_manager_->EndFrameUpdate();
 #ifdef DISABLE_EXPLICIT_SYNC
   compositor_.InsertFence(fence);
 #else
   if (fence > 0) {
-    compositor_.InsertFence(dup(fence));
-    fd_handler_.AddFd(fence);
-    out_fence_.Reset(fence);
+    if (previous_layers_.size() > 0) {
+      HWCPoll(fence, -1);
```

Comment:
Agree with @pallavigopal it will introduce additional 16ms latency, I think the tearing is due to display release fence, the app can get the previous_layers buffer ahead, but the buffer update should be done after release fence signaled.
Take the Vulkan case for example, the release fence is -1, so tearing occurs.